storage/remote/queue_manager.go

```diff
@@ -827,6 +838,7 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 
 		if err == nil {
 			s.qm.succeededSamplesTotal.Add(float64(len(samples)))
+			s.qm.bytesSent.Add(float64(len(*buf)))
```

Comment:
Nit: since we send `req` not `*buf` could we use that here as well?

Reply:
Sure. 

Do you think there's enough value in having two metrics, one using `*buf` and one usin `req`, so that we could see the difference in overhead per sample with any config changes or future streaming remote write?

Reply:
Are you suggesting tracking `cap(req) - len(req)`? If so I don't think it's worth it (yet?).

Should this be a histogram rather than a counter so that it is also possible to see the size of requests?

Reply:
I was suggesting we track the size of the data vs the size of the data wrapped up as a write request, though that actually wouldn't be as simple change as I'd have liked because of the compression.

And we're actually setting `*buf = req`.

Reply:
Ahh ok, yeah I was confused about tracking `*buf` and `req` since they are the same. I think this PR is good to start with, the nit is just that `*buf` is only being used at the top of the function which is kinda nice.